src/render/program/line_program.js

```diff
@@ -157,7 +157,11 @@ const lineDefinesValues = (layer: LineStyleLayer): LineDefinesType[] => {
     const values = [];
     if (hasDash(layer)) values.push('RENDER_LINE_DASH');
     if (layer.paint.get('line-gradient')) values.push('RENDER_LINE_GRADIENT');
-    if (layer.paint.get('line-trim-offset')) values.push('RENDER_LINE_TRIM_OFFSET');
+
+    const trimOffset = layer.paint.get('line-trim-offset');
+    if (trimOffset[0] !== 0 && trimOffset[1] !== 0) {
+        values.push('RENDER_LINE_TRIM_OFFSET');
+    }
```

Comment:
Noted that this check breaks some of the render tests https://app.circleci.com/pipelines/github/mapbox/mapbox-gl-js/10924/workflows/689b81de-634d-4e19-98f2-09f48f46ef34/jobs/143507

<img width="568" alt="Screen Shot 2022-06-20 at 19 30 13" src="https://user-images.githubusercontent.com/533564/174645337-15c9830e-22fb-4829-b322-acb1836e8597.png">

Reply:
I think the `&&` needs to be `||`